geomstats/geometry/discrete_surfaces.py

```diff
@@ -0,0 +1,313 @@
+        ----------
+        n_samples : int
+            Number of surfaces to sample.
+            Optional, Default=1
+
+        Returns
+        -------
+        vertices : array-like, shape=[n_samples, n_vertices, 3]
+            Vertices for a batch of points in the space of discrete surfaces.
+        """
+        euclidean = Euclidean(dim=3)
+        vertices = euclidean.random_point(n_samples * self.n_vertices)
+        vertices = gs.reshape(vertices, (n_samples, self.n_vertices, 3))
+        return vertices[0] if n_samples == 1 else vertices
+
+    def vertex_areas(self, point):
+        """Compute vertex areas for a triangulated surface.
+
+        Heron's formula gives the triangle's area in terms of its sides a b c:,
+        As the square root of the product s(s - a)(s - b)(s - c),
+        where s is the semiperimeter of the triangle, that is, s = (a + b + c)/2.
+
+        Vertex area is the area of all of the triangles who are in contact (incident)
+        with a specific vertex, according to the formula:
+        vertex_areas = 2 * sum_incident_areas / 3.0
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n_vertices, 3]
+             Surface, as the 3D coordinates of the vertices of its triangulation.
+
+        Returns
+        -------
+        vertex_areas :  array-like, shape=[..., n_vertices, 1]
+            Vertex area for each vertex.
+        """
+        need_squeeze = False
+        if point.ndim == 2:
+            point = gs.expand_dims(point, 0)
+            need_squeeze = True
+        n_points, n_vertices, _ = point.shape
+        vertex_0, vertex_1, vertex_2 = (
+            gs.take(point, indices=self.faces[:, 0], axis=-2),
+            gs.take(point, indices=self.faces[:, 1], axis=-2),
+            gs.take(point, indices=self.faces[:, 2], axis=-2),
+        )
+        len_edge_12 = gs.linalg.norm((vertex_1 - vertex_2), axis=-1)
+        len_edge_02 = gs.linalg.norm((vertex_0 - vertex_2), axis=-1)
+        len_edge_01 = gs.linalg.norm((vertex_0 - vertex_1), axis=-1)
+        half_perimeter = 0.5 * (len_edge_12 + len_edge_02 + len_edge_01)
+        area = gs.sqrt(
+            (
+                half_perimeter
+                * (half_perimeter - len_edge_12)
+                * (half_perimeter - len_edge_02)
+                * (half_perimeter - len_edge_01)
+            ).clip(min=1e-6)
+        )
+
+        id_vertices = gs.flatten(gs.array(self.faces))
+        id_vertices = gs.expand_dims(id_vertices, axis=0)
+        id_vertices = gs.tile(id_vertices, (n_points, 1))
+        incident_areas = gs.zeros((n_points, n_vertices))
+        val = gs.reshape(gs.tile(area, (1, 1, 3)), (n_points, -1))
+        incident_areas = gs.scatter_add(
+            incident_areas, dim=1, index=id_vertices, src=val
+        )
+        vertex_areas = 2 * incident_areas / 3.0
+
+        if need_squeeze:
+            vertex_areas = gs.squeeze(vertex_areas, axis=0)
+        return vertex_areas
```

Comment:
after create methods as suggested above, this method could simplify to:

```python
batch_shape = point.shape[:-2]
n_vertices = point.shape[-2]
n_faces = self.faces.shape[0]

area = _get_triangle_areas(point)

id_vertices = gs.broadcast_to(
    gs.flatten(self.faces),
    batch_shape + (math.prod(self.faces.shape),)
)

incident_areas = gs.zeros(batch_shape + (n_vertices,))
val = gs.reshape(
    gs.broadcast_to(gs.expand_dims(area, axis=-2), batch_shape + (3, n_faces)),
    batch_shape + (-1,),
)
incident_areas = gs.scatter_add(
    incident_areas, dim=len(batch_shape), index=id_vertices, src=val
)
vertex_areas = 2 * incident_areas / 3.0

return vertex_areas
```

So we don't really need to use `gs.squeeze`.

Reply:
I'm also avoid using `gs.tile` (by using `gs.broadcast_to` instead).